### PR TITLE
Update Media Picker example in Custom Block Editor view

### PR DESCRIPTION
**Pull request by sofietoft**
Also added the new Media Picker filter to the "Configuring Block Editor Label Properties" article.

I need help add a description to the mediaItemResolver filter in the table.

### CODE REVIEW

File: umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/block-editor/build-custom-view-for-blocks.md [hunk 1]
```diff
@@ -115,11 +115,34 @@ angular.module("umbraco").controller("customBlockController", function ($scope)
...
 
-With the setup of files above, you need to amend the `customBlock.controller.js` file, by injecting the `mediaResource` to retrieve the image from the ID:
+#### 1. Retrieve the image using a filter directly in your custom view
+
+By using the AngularJS filter called `mediaItemResolver` is it possible to retrieve the image directly from your custom view file. You will need to create a variable which then uses the `mediaItemResolver` filter on the Media Picker property. This variable is then used to retrieve the image URL in an `<img>` HTML tag.
+
+In the following example, a `mediaItem` variable is defined and used to set the `ng-src` value on the image tag. This will retrieve the image into your custom view.
+
+```html
+<div ng-controller="customBlockController" ng-click="block.edit()">
```
Comment (nielslyngsoe) [inline umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/block-editor/build-custom-view-for-blocks.md:129]: ng-controller="customBlockController" kan fjernes fra denne linje :-)

File: umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/block-editor/build-custom-view-for-blocks.md [hunk 2]
```diff
@@ -115,11 +115,34 @@ angular.module("umbraco").controller("customBlockController", function ($scope)
...
+    
+    <!-- Use the variable defined above to retrieve the mediaLink representing the image -->
+    <img ng-src=“{{mediaItem.mediaLink}}“/>
+    <p ng-bind="block.data.description"></p>
+</div>
+```
+
+#### 2. Retrive the image using your the AngularJS controller
+
+It is also possible to get the `ImageUrl` by using the unique ID in the AngularJS controller.
```
Comment (nielslyngsoe) [inline umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/block-editor/build-custom-view-for-blocks.md:143]: ville nok skrive:

using the unique ID in *an* AngularJS controller.

File: umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/block-editor/label-property-configuration.md [hunk 3]
```diff
@@ -14,15 +14,16 @@ This example will show the name of the ElementType when the value of `myProperty
...
+| -------------------------------------------------------------------------------------------------- | ------------------------------------------------------- | --------------------------- | ------------------------------------------------------------ |
+| ncNodeName                                                                                         | Gets the name of a node                                 | Umbraco node                |                                                              |
+| ncRichText                                                                                         | Strips HTML                                             | Richtext editor             |                                                              |
+| mediaItemResolver                                                                                  | ?                                                       | Media Picker                |                                                              |
```
Comment (nielslyngsoe) [inline umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/block-editor/label-property-configuration.md:21]: Description, could be:

Retrieves a Media Item object.

**Review by nielslyngsoe — COMMENTED** [review]
Lookks good, i added a few comments.. in danish 🤫

**Comment by sofietoft** [comment]
Thanks for the review @nielslyngsoe 💪 

I've made the suggested changes - let me know what you think!

File: umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/block-editor/build-custom-view-for-blocks.md [hunk 4]
```diff
@@ -115,11 +115,34 @@ angular.module("umbraco").controller("customBlockController", function ($scope)
 
 ### Example: Displaying an image from a Media Picker
 
-Your block may enable you to 'pick' an image for use as the background for a particular block. If you try to display this image directly in the view using `block.data.image`, you will see the unique ID and not the image.
+Your block may enable you to 'pick' an image to use as the background for a particular block. If you try to display this image directly in the view using `block.data.image`, you will see the unique ID and not the image.
 
-You will need to use the ID in our custom AngularJS controller in order to get the `ImageUrl` to display in our backoffice Block Editor View.
+There are two approaches to display the actual image in your custom block view. Either use a filter in your custom view or retrieve the image through the AngularJS controller.
```
Comment (eshanrnh) [inline umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/block-editor/build-custom-view-for-blocks.md:120]:
```suggestion
There are two approaches to displaying the actual image in your custom block view. Either use a filter in your custom view or retrieve the image through the AngularJS controller.
```

File: umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/block-editor/build-custom-view-for-blocks.md [hunk 5]
```diff
@@ -115,11 +115,34 @@ angular.module("umbraco").controller("customBlockController", function ($scope)
...
 
-With the setup of files above, you need to amend the `customBlock.controller.js` file, by injecting the `mediaResource` to retrieve the image from the ID:
+#### 1. Retrieve the image using a filter directly in your custom view
+
+By using the AngularJS filter called `mediaItemResolver` is it possible to retrieve the image directly from your custom view file. You will need to create a variable which then uses the `mediaItemResolver` filter on the Media Picker property. This variable is then used to retrieve the image URL in an `<img>` HTML tag.
```
Comment (eshanrnh) [inline umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/block-editor/build-custom-view-for-blocks.md:124]:
```suggestion
By using the AngularJS filter called `mediaItemResolver`, it is possible to retrieve the image directly from your custom view file. You will need to create a variable that then uses the `mediaItemResolver` filter on the Media Picker property. This variable is then used to retrieve the image URL in an `<img>` HTML tag.
```

File: umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/block-editor/build-custom-view-for-blocks.md [hunk 6]
```diff
@@ -115,11 +115,34 @@ angular.module("umbraco").controller("customBlockController", function ($scope)
...
+    <!-- Use the 'mediaItemResolver' on the block.data.image -->
+    {{mediaItem = (block.data.image[0].mediaKey | mediaItemResolver); “”}}
+
+    <!-- Use the variable defined above to retrieve the mediaLink representing the image -->
+    <img ng-src=“{{mediaItem.mediaLink}}“/>
+    <p ng-bind="block.data.description"></p>
+</div>
+```
+
+#### 2. Retrive the image using your the AngularJS controller
```
Comment (eshanrnh) [inline umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/block-editor/build-custom-view-for-blocks.md:141]:
```suggestion
#### 2. Retrieve the image using your AngularJS controller
```

File: umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/block-editor/label-property-configuration.md [hunk 7]
```diff
@@ -14,15 +14,16 @@ This example will show the name of the ElementType when the value of `myProperty
...
+| ncRichText                                                                                         | Strips HTML                                             | Richtext editor             |                                                              |
+| mediaItemResolver                                                                                  | Retrieves a Media Item object                           | Media Picker                |                                                              |
+| [limitTo](https://docs.angularjs.org/api/ng/filter/limitTo)                                        | AngularJS native truncate                               | String                      | n: maximum length of the string                              |
+| [truncate](https://apidocs.umbraco.com/v10/ui/#/api/umbraco.filters.filter:truncate)               | Umbraco's richer truncate function                      | String                      | wordwise: boolean to indicaste whether to truncate a string mid-word or not<br />max: maximum length of the string<br />tail (optional): string to indicate a truncated string, "`…`" by default |
```
Comment (eshanrnh) [inline umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/block-editor/label-property-configuration.md:23]:
```suggestion
| [truncate](https://apidocs.umbraco.com/v10/ui/#/api/umbraco.filters.filter:truncate)               | Umbraco's richer truncate function                      | String                      | wordwise: boolean to indicate whether to truncate a string mid-word or not<br />max: maximum length of the string<br />tail (optional): string to indicate a truncated string, "`…`" by default |
```

**Review by eshanrnh — CHANGES_REQUESTED** [review]
Looks good, @sofietoft 💪 Added a couple of comments

**Pull request merged**